parser/openfoodfacts_taxonomy_parser/parser/parser.py

```diff
@@ -234,14 +235,37 @@ def _write_to_database(self, taxonomy: Taxonomy, taxonomy_name: str, branch_name
         self._create_child_links(taxonomy.child_links, project_label)
         self._create_previous_links(taxonomy.previous_links, project_label)
 
-    def __call__(self, filename: str, branch_name: str, taxonomy_name: str):
+    def __call__(
+        self,
+        main_filename: str,
+        external_filenames: list[str] | None,
+        branch_name: str,
+        taxonomy_name: str,
+    ):
         """Process the file"""
         start_time = timeit.default_timer()
 
         branch_name = normalize_text(branch_name, char="_")
         taxonomy_parser = TaxonomyParser()
         try:
-            taxonomy = taxonomy_parser.parse_file(filename, self.parser_logger)
+            if external_filenames is None:
+                taxonomy = taxonomy_parser.parse_file(main_filename, self.parser_logger)
+            else:
+                # parse external taxonomies files if any, and add their entry nodes to the main taxonomy
+                external_entry_nodes = []
+                for filename in external_filenames:
+                    external_taxonomy_parser = TaxonomyParser()
+                    external_taxonomy = external_taxonomy_parser.parse_file(
+                        filename, self.parser_logger
+                    )
+                    external_entry_nodes.extend(external_taxonomy.entry_nodes)
+                for entry_node in external_entry_nodes:
+                    entry_node.is_external = True
+                # parse main taxonomy file
+                taxonomy = taxonomy_parser.parse_file(
+                    main_filename, self.parser_logger, external_entry_nodes
+                )
```

Comment:
What do you think if putting that as part of the `TaxonomyParser` (it would call itself recursively if there are external taxonomies)? 
The reason I am suggesting this is that I feel that we have two main concerns:
- parsing the `.txt` file into a Taxonomy AST
- transforming the Taxonomy AST into Cypher queries to be run to write the taxonomy to the database
And I feel that this code does a part of the Taxonomy AST creation